pkg/webhook/resourceinterpretercustomization/validating.go

```diff
@@ -0,0 +1,55 @@
+package resourceinterpretercustomization
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+)
+
+// Check if our ValidatingAdmission implements necessary interface
+var _ admission.Handler = &ValidatingAdmission{}
+var _ admission.DecoderInjector = &ValidatingAdmission{}
+
+// ValidatingAdmission validates ResourceInterpreterCustomization object when creating/updating.
+type ValidatingAdmission struct {
+	client.Client
+	decoder *admission.Decoder
+}
+
+// Handle implements admission.Handler interface.
+// It yields a response to an AdmissionRequest.
+func (v *ValidatingAdmission) Handle(ctx context.Context, req admission.Request) admission.Response {
+	configuration := &configv1alpha1.ResourceInterpreterCustomization{}
+
+	err := v.decoder.Decode(req, configuration)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	klog.V(2).Infof("Validating ResourceInterpreterCustomization(%s) for request: %s", configuration.Name, req.Operation)
+	configs := &configv1alpha1.ResourceInterpreterCustomizationList{}
+	if err = v.List(ctx, configs); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	for _, config := range configs.Items {
+		if config.Name != configuration.Name &&
+			config.Spec.Target.APIVersion == configuration.APIVersion &&
+			config.Spec.Target.Kind == configuration.Kind &&
+			!validateCustomizationRule(config.Spec.Customizations, configuration.Spec.Customizations) {
```

Comment:
This judgement may conflict with what was said at today's community meeting: `InterpretDependency` operation can support multiple different script definitions.

I think it's possible to extract the whole judgment condition into a method.

Reply:
done!